### PR TITLE
More some more code to generic well model

### PR DESCRIPTION
**Pull request by akva2**
Reorganizations to enable moving more code into the Generic base class

### CODE REVIEW

**Comment by akva2** [comment]
jenkins build this please

File: opm/simulators/flow/BlackoilModelNldd.hpp [hunk 1]
```diff
@@ -815,7 +815,7 @@ class BlackoilModelNldd {
                            const SimulatorTimerInterface& timer,
                            const Domain& domain)
     {
-        auto initial_local_well_primary_vars = model_.wellModel().getPrimaryVarsDomain(domain);
+        auto initial_local_well_primary_vars = model_.wellModel().getPrimaryVarsDomain(domain.index);
```
Comment (atgeirr) [inline opm/simulators/flow/BlackoilModelNldd.hpp:818]: Why did you change the interface from taking a Domain to taking an index?

Comment (akva2) [reply opm/simulators/flow/BlackoilModelNldd.hpp:818]: because Domain is dependent on the full typetag.

**Comment by akva2** [comment]
jenkins build this please

**Review by bska — APPROVED** [review]
Looks good to me.  Please merge when green.

**Comment by bska** [comment]
Build check is green.  I'll merge into master.
